disnake/state.py

```diff
@@ -1880,6 +1881,31 @@ def parse_auto_moderation_action_execution(
         event = AutoModActionExecution(data=data, guild=guild)
         self.dispatch("automod_action_execution", event)
 
+    def parse_guild_audit_log_entry_create(self, data: gateway.AuditLogEntryCreate) -> None:
+        guild = self._get_guild(int(data["guild_id"]))
+        user: Optional[User] = None
+        if user_id := data["user_id"]:
+            user = self.get_user(int(user_id))
+        if guild is None:
```

Comment:
Should I `return` if the user isn't found in the cache?

Also, I think a raw event for this might be useful. 🤔

Reply:
> A raw event could be useful, though mainly for those that disable `Intents.guilds`.

Hmm, from my testing, Intents.guilds is required in order to receive the event 🤔

Reply:
Ah, gotcha. In that case, is there a point in having a raw event? Since audit logs generally fall back to `Object` when something isn't cached, the non-raw event should be fine.

Reply:
> Instead of getting that specific user from the cache, it would be easier to pass the entire `self._users` dict into `AuditLogEntry`. Could do the same with the guild's `guild_scheduled_events` and `threads`, the others unfortunately aren't cached.

changed in 40d5169dc769565db0adf867e02327172890463c